src/modules/questions/questions.service.ts

```diff
@@ -20,47 +26,112 @@ export class QuestionsService {
     private readonly questionOptionRepository: Repository<QuestionOption>,
     @Inject(CACHE_MANAGER)
     private readonly cacheManager: Cache,
+    private readonly testsService: TestsService,
+    private readonly dataSource: DataSource,
   ) {}
 
   async create(createQuestionDto: CreateQuestionDto, authContext: AuthContext): Promise<Question> {
-    const { options, ...questionData } = createQuestionDto;
+    const { options, testId, sectionId, isCompulsory, ...questionData } = createQuestionDto;
     
     // Validate question data
-    this.validateQuestionData(createQuestionDto);
-    
-    // Validate question options (includes duplicate validation)
+    this.validateQuestionData(createQuestionDto, authContext);
```

Comment:
@vinayak-patil validate question data, params etc after validating test and section if provided.